pkg/healthcheck/healthcheck.go

```diff
@@ -1395,6 +1424,41 @@ func (hc *HealthChecker) allCategories() []*Category {
 	}
 }
 
+// CheckProxyVersionsUpToDate checks if all the proxies are on the latest
+// installed version
+func (hc *HealthChecker) CheckProxyVersionsUpToDate(pods []corev1.Pod) error {
+	return CheckProxyVersionsUpToDate(pods, hc.LatestVersions)
+}
+
+// CheckProxyVersionsUpToDate checks if all the proxies are on the latest
+// installed version
+func CheckProxyVersionsUpToDate(pods []corev1.Pod, versions version.Channels) error {
+	outdatedPods := []string{}
+	for _, pod := range pods {
+		proxyVersion := k8s.GetProxyVersion(pod)
+		if err := versions.Match(proxyVersion); err != nil {
+			outdatedPods = append(outdatedPods, fmt.Sprintf("\t* %s (%s)", pod.Name, proxyVersion))
+		}
+	}
+	if len(outdatedPods) > 0 {
+		podList := strings.Join(outdatedPods, "\n")
+		return fmt.Errorf("Some data plane pods are not running the current version:\n%s", podList)
```

Comment:
This will require rewording because the check is now used in the control plane, not just the data plane. Something like just "Some proxies are not running the current version: \n%s".